source/agora/common/BanManager.d

```diff
@@ -253,11 +259,31 @@ public class BanManager
 
     public void banUntil (Address address, TimePoint banned_until) @safe nothrow
     {
-        if (this.get(address).whitelisted)
+        if (address in this.whitelisted)
             return; // Whitelisted address
 
         log.info("BanManager: Address {} banned until {}", address, banned_until);
-        this.get(address).banned_until = banned_until;
+        try
+        {
+            this.url_failures.require(address).banned_until = banned_until;
+        }
+        catch (Exception e)
+        {
+            log.warn("Exception setting banUntil for address {}", address);
+        }
+    }
+
+    private void storeBanned (Address address, TimePoint banned_until) @trusted nothrow
```

Comment:
Couldn't find any calls to this

Reply:
Well spotted. Will add now.

Reply:
In fact I'll add some unit tests for the database also.